
# Analysis of bulk RNA-seq from zebrafish

This is a snakemake pipeline and jupyter notebook for the analysis of RNA-seq data generated from zebrafish cardiomyotes, prepared and sequenced with CEL-seq protocol.
The pipeline and downstream analysis consist of:

* Trimming adaptors from fastq files
* QC of fastq read quality with [fastqc](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
* Alignment of transcriptomic reads with [STAR](https://github.com/alexdobin/STAR)
* Count matrix creation with [Rsubread featureCounts](https://bioconductor.org/packages/release/bioc/html/Rsubread.html)
* Differential expression analysis with [DESeq2](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html)

## Project structure

### directories
data/ contains all processed data generated from the analysis pipeline (main.snake) e.g. fastq, bam

notebooks/ are the records of interactive analyses of data generated by the pipeline

raw/ raw data from the sequencer that is not to be touched

scripts/ functions or bits of code used in main pipeline or analysis notebooks e.g. snakemake rules or R functions

### files
* main.snake is the master snakemake file for the standard mapping pipeline
* [conda_setup.md](conda_setup.md) gives a description of the set up of environment variables
  * pansnakemake_environment.yml is the conda environment used for the snakemake pipeline
  * pannotebooks_environment.ymlis the conda environment used for the jupyter notebooks
* samplesheet.tsv contains all sample names and fastq paths
* samplesheet_aligning.tsv contains paired information for samples paired-end sequenced
* snake_run.sh script for job submission to the Grid Engine cluster
