# Analysis of CEL-seq3 data generated by Daniela Panakovas lab

## Data generation and sequencing
* Samples were sorted by Anne Merks before lockdown (Feb-March 2020)
* Ronny made libraries finished 26/05/2020
* Went on sequencer run by Anika finished 4/06/2020
* Anika mapped and demultiplexed files /fast/AG_Junker/aneusch/dataExchange/20200605_bulkSeq_RonnyAlison/datain and I stored them in /fast/AG_Ohler/Alison/panakova_CM_RNAseq/raw


## Demultiplexing issue
After Anika demultiplexed, many barcodes had few reads, and there were some "barcodes" that dont match our barcodes with large number of reads "*strange_pasted*"

```
du -h -c /fast/AG_Ohler/Alison/panakova_CM_RNAseq/raw/*fastq
```

gives 23G
```
/fast/AG_Ohler/Alison/panakova_CM_RNAseq/raw/*strange*fastq
```
gives 12G

=> worth trying to find unmatched barcodes.

### barcode matching 12/06/2020

R1_pasted_strange_ACGGCACG.fastq
matches
sample 19: ACGGCACC with one G swap at the end

All other "strange" share some overlap with each other
  ATCGTCGG
CGATCGTC
       CGGACTGT
    CGTCGGAC
 GATCGTCG
     GTCGGACT
   TCGTCGGA
and give a total assembled sequence of:
CGATCGTCGGACTGT
Complement: CTAGCAGCCTGACA
Reverse: TGTCAGGCTGCTAG
Reverse complement: ACAGTCCGACGATC
This matches the last part of the read1 sequencing primer (at least according to this resource https://teichlab.github.io/scg_lib_structs/methods_html/CEL-seq_family.html). 
So it suggests to me that something went wrong in the priming of R1 (in the case of CEL-seq3 barcode is in R2)


In theory we could estimate how much the bases are shifted by the sequence we were got in the "R1_pasted_strange_" files, assuming that the primer bound in a position upstream of where it should have. So in file R1_pasted_strange_GATCGTCG.fastq the barcode should be the following 8bp. But in R1_pasted_strange_CGGACTGT.fastq you would need to move 6bp downstream then the barcode should start.

### 17/06/2020 grep for barcodes in unmapped reads

In notebook file:
* Filtered all R2 reads (which should cover barcode) for the "strange" barcodes that Anika found.
* Did a grep on these for known barcodes
- found that they are the reverse complement of R1 sequencing barcodes from CEL-seq
- only 7bp non barcode reads are present in R1_pasted_strange_CGATCGTC_R2.fastq and as you shift right (in their alignment) fewer non-barcode reads are present to the point of 0 in R1_pasted_strange_CGGACTGT_R2.fastq

-> So probably we cant identify barcodes from these or only from R1_pasted_strange_CGATCGTC_R2.fastq, R1_pasted_strange_R1_pasted_strange_CGATCGTC_R2.fastq_R2.fastq
To do
=> find out if R2 primers in CEL-seq3 are same as R1 primers in CEL-seq2? (why would you reverse transcribe these?)
=> find out if there are UMIs in CEL-seq3, and if these are upstream of the primer sequence we wont be able to find the barcodes

### 30/06/2020

DESeq2 between bleb vs dmso and vangl2 vs wt

### 01/07/2020
Check that all the samples with low% alignment are the ones excluded

Looking again at the sample qualities - I should exclude strange sample 19
- Only 9.9% aligned (suspicious)
- In teh PCA plot it clusters more closely with the dmso samples

Try trimming
- Itai Yanai pipeline doesnt trim

Illumina RA3 adapter
TGGAATTCTCGGGTGCCAAGG

-> Trimming did improve the sequence quality and GC content
-> But for the good samples this made relatively little different to the mapping

Rerun the notebook for mapped reads

Why does dmso sample1 cluster with vangl2? Checked all the barcodes and match the sample names

Sequencing increased depth might not help for successful samples becuase the duplication rate seems to be higher in these (probably low complexity starting point)

### 02/07/2020 

Sent Anne and Daniela the results 

* DESeq2_trimmed_notebook_200702.ipynb is the first analysis version sent to Daniela and Anne on 02/07/2020 by email "Cardiomyocyte RNA-seq analysis results 1" 
	- Freeze this - next time I do more analysis, copy this to a new notebook and proceed
	- Outputs are in deseq2_trimmed_outputs_200702 and deseq2_trimmed_outputs_200702.zip

Sent Anne the plots for GO terms (that she already had the enrichment results, just wanted to make a nicer plot)
* GO_term_plot.ipynb

### 31/07/2020

Tried running all the macrogen and Ronnys data together, but get fewer DE genes that if run separately notebook: DESeq2_trimmed_macrogen_data.ipynb

Tried running macrogen data separately from Ronnys data (but normalising Blebb and Vangl2 experiments together): DESeq2_trimmed_macrogen_data_alone.ipynb
    - see poor overlap in Vangl2 results compared to macrogen, and lack of down regulation of Vangl2 suggest that maybe blebb is dominating variance normalisation

Try running vangl2 data separately from Blebb data


Mismatches between macrogen analysis and mine:
- Total number of reads - they give around 40M per sample (separate by lanes), I got aroun 20M (but mine are from separated samples by lane. Cumulatively they add up to same).
- What are samples CTL30hpf1 compared to cmCTL30hpf1?
- They see ~85% mapping - I get around 90%
- I used STAR for mapping, they used Tophat2
- They used cufflinks for read count quantification and I used Rsubread (cufflinks shouldnt be used for DESeq2 input, check whether they did)
- For differential expression they have incorporatied 3 batches as effects?
- They describe using the CTL30hpf1 data rather than cmCTL30hpf1 data
- They filtered out genes where more than one sample had FPKM of 0
- log2 + 1 raw signal normalise (but they call the FPKM raw, which it shouldnt be) and quantile normalise
- No use of p-values, just do fold change
Main questions
-> am I comparing the same data?
-> what is their statistical test, just fold change in more than one condition?
-> should I be applying a batch correction?

### 24/08/2020

When I reanalysed the bulk RNA-seq data that had been generated by macrogen I normalised the blebb, dmso, vangl2 and wt samples all together and then ran DE analysis on blebb vs dmso and vangl2 vs wt. However the vangl2 had very few DE genes and these had very little overlap with the bleb indiced genes. 
Tried exploring this further and found that if I separated blebb and dmso fromt eh vangl2 and wt samples for the normalisation I got more DE genes from vangl2 that match better with dmso. 
Sent this data back to Anne.

### 22/09/2020

After re-ran the analysis with the blebb samples separated from the vangl2 samples, the DE genes matched expected GO terms for muscle activity
=> normalising a highly variable sample with a less variable sample has the risk of flattening the variation of the less variant sample.

Made vaolcano plots, euclidean distance matrices, GO term plots and lists of all genes with logFC and p-value for the macrogen data (normal bulk RNA-seq) and the CELseq (low input) data for blebb vs dmso and for vangl2 mutant for Annes manuscript.
